praw/models/reddit/mixins/exportable.py

```diff
@@ -0,0 +1,37 @@
+"""Provide the ExportableMixin class."""
+from typing import Any, Callable, Dict, TypeVar, Union
+
+_T = TypeVar("_T")
+Reddit = TypeVar("Reddit")
+
+
+class ExportableMixin:
+    """A mixin for classes that can be easily exported and re-created."""
+
+    @staticmethod
+    def _create_base(
+        base: Callable[
+            [Reddit, str, Dict[str, Union[Any, Dict[str, Any]]]], _T
+        ],
+        reddit: Reddit,
+        data: Union[str, Dict[str, Union[Any, Dict[str, Any]]]],
+    ) -> _T:
+        """Create another PRAWBase with either a string or a dict."""
+        return (
+            base(reddit, _data=data)
+            if isinstance(data, dict) and ("_")
+            else base(reddit, data)
+        )
+
+    def export(self) -> Dict[str, Union[Any, Dict[str, Any]]]:
+        """Export the data of the class, as a dict of the attributes.
+
+        Other ExportableMixin classes are exported as dicts.
+
+        The exported data should be able to re-create a copy of the class
+        that is identical to the class the data was exported from.
+        """
+        return {
+            key: value.export() if hasattr(value, "export") else value
+            for key, value in self.__dict__.copy().items()
```

Comment:
In my reading, this won't export a JSON-serializable dict, because it will end up exporting `Reddit` instances with `key`s like `_reddit`. My understanding was that the point of this feature was to allow objects to be saved as JSON and then re-instantiated, and so I *think* this would break that.

Reply:
I am working on a version that actually does do that, which I haven’t pushed yet.